DocumentDBStudio/AccountSettingsForm.cs

```diff
@@ -105,10 +114,43 @@ private void btnOK_Click(object sender, EventArgs e)
             Settings.Default.Save();
         }
 
+        private void ProcessTokensString(string tokenText, string collections)
+        {
+            //format type=resource=token;associatedColection;otherTokens
+            this.AccountSettings.collectionTokens = new List<string>();
+            this.AccountSettings.collections = new List<string>();
+            string tokenBeggining = "type=resource";
+            List<int> tokenStartingPoints = AllIndexesOf(tokenText, tokenBeggining);
+            string[] collectionSplit = collections.Split(';');
+            tokenStartingPoints.Add(tokenText.Length);
+            for (int i = 0 ; i < tokenStartingPoints.Count - 1 ; i++)
```

Comment:
This all looks pretty complex for what seems like it could be simple.

Could you this in a more readable way, and therefore easier to see if the code is right, using Linq an iterating over the substrings from split rather then using your own for loop and many int's representing string indexes?